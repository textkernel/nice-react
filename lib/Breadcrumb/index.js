export { default as BreadcrumbGroup } from './BreadcrumbGroup';
export { default as BreadcrumbItem } from './BreadcrumbItem';