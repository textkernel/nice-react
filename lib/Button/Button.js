import React, { Component } from 'react';
import PropTypes from 'prop-types';
import niceProps from '../props';

export default class Button extends Component {
  handleClick(e) {
    e.preventDefault();
    const { onClick } = this.props;
    if (!onClick) {
      return false;
    }
    onClick();
    return true;
  }

  render() {
    const { children, context, size } = this.props;
    const classes = ['nice-btn', context, size].filter(c => c && c.length > 0);

    return (
      <button type="button" onClick={ e => this.handleClick(e) } className={classes.join(' ')}>
        {children}
      </button>
    );
  }
}

Button.propTypes = {
  context: PropTypes.oneOf(niceProps.buttonContexts),
  size: PropTypes.oneOf(niceProps.buttonSizes),
  children: PropTypes.node,
  onClick: PropTypes.func,
};

Button.defaultProps = {
  context: 'primary',
  size: 'medium',
  children: null,
  onClick: null
};
