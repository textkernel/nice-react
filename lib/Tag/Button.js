import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const Button = ({ icon, className, onClick, ...rest }) => {
    const handleClick = (event) => {
        event.stopPropagation();

        if (onClick) {
            onClick(event);
        }
    };

    return (
        <button
            className={classnames('nice-tag-btn', className)}
            onClick={handleClick}
            {...rest}
        >
            <i className={`icon ${icon}`} />
        </button>
    );
};


Button.propTypes = {
    icon: PropTypes.string.isRequired,
    className: PropTypes.string,
    onClick: PropTypes.func,
};


Button.defaultProps = {
    className: '',
    onClick: null,
};


export default Button;