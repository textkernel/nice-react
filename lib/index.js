import '@textkernel/nice/dist/nice.min.css';

export { default as Panel } from './Panel';
export { Button, ButtonGroup } from './Buttons';
export { default as Progress } from './Progress';
export { default as Loading } from './Loading';
export { BreadcrumbGroup, BreadcrumbItem } from './Breadcrumb';