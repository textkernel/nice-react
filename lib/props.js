const contexts = ['brand', 'link', 'primary', 'accent', 'good', 'warning', 'bad'];
const sizes = ['small', 'medium', 'large'];
 
const props = {
  contexts,
  buttonContexts: contexts,
  buttonSizes: sizes 
};

module.exports = props;