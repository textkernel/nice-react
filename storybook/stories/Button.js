import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import { Button } from 'nice-react';

storiesOf('Button', module)
  .add('Default', () => (
    <Button onClick={() => alert('Default')}>Default</Button>
  ))
  .add('Primary', () => (
    <Button context="primary">Primary</Button>
  ))
  .add('Large', () => (
    <Button size="large">Large</Button>
  ))
  .add('Small', () => (
    <Button size="small">Small</Button>
  ));;