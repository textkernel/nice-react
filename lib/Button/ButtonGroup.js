import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import props from '../props';

const ButtonGroup = ({ children, justified, stacked, size, pullRight }) => (
    <div
        className={ classNames('nice-btn-group', {
            justified,
            stacked,
            [size]: size,
            'pull-right': pullRight
        }) }
    >
        { children }
    </div>
);

ButtonGroup.propTypes = {
    children: PropTypes.node,
    size: PropTypes.oneOf(props.buttonGroupSizes),
    justified: PropTypes.bool,
    stacked: PropTypes.bool,
    pullRight: PropTypes.bool
};

ButtonGroup.defaultProps = {
    size: null,
    children: null,
    justified: false,
    stacked: false,
    pullRight: false
};

export default ButtonGroup;
