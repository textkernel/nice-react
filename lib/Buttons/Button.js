import React, { Component } from 'react';
import PropTypes from 'prop-types';
import props from '../props';

export default class Button extends Component {
  handleClick(e) {
    e.preventDefault();
    this.props.onClick();
  }

  render() {
    const { children, icon, alternative } = this.props;
    const iconClass = icon ? `icon ${icon}` : null;

    const regularButton = () => {
      const { context, size } = this.props;
      const baseClass = 'nice-btn';
      return [baseClass, context, size, iconClass];
    };

    const alternativeButton = () => {
      const baseClass = 'nice-btn-alt';
      return [baseClass, iconClass];
    };

    const butonClasses = alternative ? alternativeButton : regularButton;
    const classes = butonClasses().filter(c => c && c.length > 0);

    return (
      <a href='#' onClick={e => this.handleClick(e)} className={classes.join(' ')}>
        {children}
      </a>
    );
  }
}

Button.propTypes = {
  context: PropTypes.oneOf(props.buttonContexts),
  size: PropTypes.oneOf(props.buttonSizes),
  children: PropTypes.node,
  onClick: PropTypes.func,
  icon: PropTypes.string,
  alternative: PropTypes.bool
};

Button.defaultProps = {
  context: 'default',
  size: 'medium',
  children: null,
  onClick: null,
  icon: null,
  alternative: false
};