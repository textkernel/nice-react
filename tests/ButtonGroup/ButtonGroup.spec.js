import React from 'react';
import { Button, ButtonGroup } from '../../lib/Button';

describe('<ButtonGroup />', () => {
    it('should render the Buttons inside ButtonGroup', () => {
        const wrapper = shallow(<ButtonGroup>
                <Button>
                        Button 1
                </Button>
                <Button>
                        Button 2
                </Button>
        </ButtonGroup>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a justified ButtonGroup', () => {
        const wrapper = shallow(<ButtonGroup justified>
                <Button href="#">Some</Button>
                <Button href="#">Justified</Button>
                <Button href="#">Buttons</Button>
        </ButtonGroup>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a stacked ButtonGroup', () => {
        const wrapper = shallow(<ButtonGroup stacked>
                <Button>Some</Button>
                <Button>Stacked</Button>
                <Button>Buttons</Button>
        </ButtonGroup>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a small ButtonGroup', () => {
        const wrapper = shallow(<ButtonGroup small>
                <Button>Some</Button>
                <Button>Stacked</Button>
                <Button>Buttons</Button>
        </ButtonGroup>);
        expect(wrapper).toMatchSnapshot();
    });
});
