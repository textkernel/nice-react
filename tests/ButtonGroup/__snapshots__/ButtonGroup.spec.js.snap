// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonGroup /> should render a justified ButtonGroup 1`] = `
<div
  className="nice-btn-group justified"
>
  <Button
    alternative={false}
    context="default"
    href="#"
    icon={null}
    onClick={null}
    pullRight={false}
    size="medium"
  >
    Some
  </Button>
  <Button
    alternative={false}
    context="default"
    href="#"
    icon={null}
    onClick={null}
    pullRight={false}
    size="medium"
  >
    Justified
  </Button>
  <Button
    alternative={false}
    context="default"
    href="#"
    icon={null}
    onClick={null}
    pullRight={false}
    size="medium"
  >
    Buttons
  </Button>
</div>
`;

exports[`<ButtonGroup /> should render a small ButtonGroup 1`] = `
<div
  className="nice-btn-group"
>
  <Button
    alternative={false}
    context="default"
    href={null}
    icon={null}
    onClick={null}
    pullRight={false}
    size="medium"
  >
    Some
  </Button>
  <Button
    alternative={false}
    context="default"
    href={null}
    icon={null}
    onClick={null}
    pullRight={false}
    size="medium"
  >
    Stacked
  </Button>
  <Button
    alternative={false}
    context="default"
    href={null}
    icon={null}
    onClick={null}
    pullRight={false}
    size="medium"
  >
    Buttons
  </Button>
</div>
`;

exports[`<ButtonGroup /> should render a stacked ButtonGroup 1`] = `
<div
  className="nice-btn-group stacked"
>
  <Button
    alternative={false}
    context="default"
    href={null}
    icon={null}
    onClick={null}
    pullRight={false}
    size="medium"
  >
    Some
  </Button>
  <Button
    alternative={false}
    context="default"
    href={null}
    icon={null}
    onClick={null}
    pullRight={false}
    size="medium"
  >
    Stacked
  </Button>
  <Button
    alternative={false}
    context="default"
    href={null}
    icon={null}
    onClick={null}
    pullRight={false}
    size="medium"
  >
    Buttons
  </Button>
</div>
`;

exports[`<ButtonGroup /> should render the Buttons inside ButtonGroup 1`] = `
<div
  className="nice-btn-group"
>
  <Button
    alternative={false}
    context="default"
    href={null}
    icon={null}
    onClick={null}
    pullRight={false}
    size="medium"
  >
    Button 1
  </Button>
  <Button
    alternative={false}
    context="default"
    href={null}
    icon={null}
    onClick={null}
    pullRight={false}
    size="medium"
  >
    Button 2
  </Button>
</div>
`;
